### PR TITLE
Adding with-sce-script-dir flag to configure.ac. Fixes #1016

### LINKED ISSUE #1016: Configure flag for sce script directory
The SCE feature currently stores and executes shell scripts in /tmp. Some security benchmarks require /tmp to be non-executable, causing the scan to fail.

A solution would be to add a parameter to the configure.ac to customize the directory used by SCE for storage and execution of scripts.

### PR DESCRIPTION
**Pull request by JennyUWaterloo**
When configure is run with --enable-sce, add a new parameter --with-sce-script-dir to set the directory path for SCE to store and execute script files (default /tmp). Configure will append -DSCE_SCRIPT_DIR to CFLAGS (P_tmpdir is always defined by stdio.h, which is included by curl.h. Therefore we added SCE_SCRIPT_DIR to prevent the provided cflag from being overwritten.)

### CODE REVIEW

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by JennyUWaterloo** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci add to whitelist

File: configure.ac [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ AC_ARG_WITH([sce-script-dir],
 
 if test "x${sce}" = xyes; then
   AC_DEFINE([ENABLE_SCE], [1], [compilation of script check engine enabled])
-  CFLAGS="$CFLAGS -DSCE_SCRIPT_DIR=\"${with_sce_script_dir}\""
+  CFLAGS="$CFLAGS -DSCE_SCRIPT_DIR=\\\"${with_sce_script_dir}\\\""
```
Comment (matejak) [inline configure.ac]: Please add a comment why the double escape is needed, it is not obvious.

Comment (JennyUWaterloo) [reply configure.ac]: It's not obvious to me either. The project compiled on my ubuntu container with single escaped quotes, but failed to compile on CI (https://jenkins.open-scap.org/job/openscap-pull-requests/612/console - ```error: 'tmp' undeclared```) until the quotes were double escaped.

Comment (matejak) [reply configure.ac]: This is interesting. I am not 100% sure about what the cause may be. People are quite confused in this discussion too:
https://stackoverflow.com/questions/2410976/how-to-define-a-string-literal-in-gcc-command-line/15227471
The more robust, but also more arcane solution would be to write some macro magic to stringify the incoming macro, but I am fine if you just include a comment saying that the double escape is a tricky manner, very likely required in connection with jenkins CI.
Last but not least, please add the `configure.ac` code changes to `ac_probes/configure.ac.tpl` (I usually use `vimdiff` to find the right place where to add the lines.)

**Comment by JennyUWaterloo** [comment]
@matejak I've applied the requested changes

**Comment by matejak** [comment]
@JennyUWaterloo thank you for your great observation! I see that something like `OSCAP_TEMP_DIR` should work fine, so feel free to go ahead with the rename.

**Comment by matejak** [comment]
Great contribution, thank you for your perseverance! It might have been frustrating that this PR was not so straightforward, but it was all for the good reasons. Thank you!

**Comment by mpreisler** [comment]
@jan-cerny this will need a careful merge into master and we will need to redo this with cmake.

**Comment by jan-cerny** [comment]
@mpreisler That's awesome! Thanks.

**Pull request merged**